src/library/libraryfeature.cpp

```diff
@@ -6,18 +6,37 @@
 // KEEP THIS cpp file to tell scons that moc should be called on the class!!!
 // The reason for this is that LibraryFeature uses slots/signals and for this
 // to work the code has to be precompiles by moc
-LibraryFeature::LibraryFeature(QObject *parent)
-        : QObject(parent) {
 
+LibraryFeature::LibraryFeature(
+        QObject *parent)
+        : QObject(parent),
+          m_clickedChildActivationTimeoutMillis(0) {
 }
 
-LibraryFeature::LibraryFeature(UserSettingsPointer pConfig, QObject* parent)
+LibraryFeature::LibraryFeature(
+        int clickedChildActivationTimeoutMillis,
+        QObject *parent)
         : QObject(parent),
-          m_pConfig(pConfig) {
+          m_clickedChildActivationTimeoutMillis(clickedChildActivationTimeoutMillis) {
+    DEBUG_ASSERT(m_clickedChildActivationTimeoutMillis >= 0);
 }
 
-LibraryFeature::~LibraryFeature() {
+LibraryFeature::LibraryFeature(
+        UserSettingsPointer pConfig,
+        QObject* parent)
+        : QObject(parent),
+          m_pConfig(pConfig),
+          m_clickedChildActivationTimeoutMillis(0) {
```

Comment:
Why is 0 here allowed?

Reply:
So the timer fires mediately with 0? The Qt doc is not clear about that, or negative values.

Reply:
Totally clear about that: "As a special case, a QTimer with a timeout of 0 will time out as soon as all the events in the window system's event queue have been processed. This can be used to do heavy work while providing a snappy user interface: ..."